packages/block-library/src/post-comments-count/edit.js

```diff
@@ -59,15 +71,11 @@ export default function PostCommentsCountEdit( {
 					} }
 				/>
 			</BlockControls>
-			<div { ...blockProps }>
-				{ postId && commentsCount !== undefined ? (
-					commentsCount
-				) : (
-					<Warning>
-						{ __( 'Post Comments Count block: post not found.' ) }
-					</Warning>
-				) }
-			</div>
+			{ postId && commentsCount !== undefined ? (
+				<div { ...blockProps }>{ commentsCount }</div>
+			) : (
+				<NoCommentsWarning { ...blockProps } />
```

Comment:
Interestingly for some reason I _think_ this change means that I can no longer select the block in the site editor. Prior to this PR, if I click the placeholder, I can select the block. With this PR, if I click on the placeholder, nothing seems to happen. I wonder if this means that the `...blockProps` needs to be expanded directly on a div element rather than a React component? I'm not too sure how the block selection magic works 🤔

| Before | After |
| --- | --- |
| ![2022-08-23 15 02 49](https://user-images.githubusercontent.com/14988353/186074072-22d5d2ed-ddee-4a1b-8c86-86e79923b3b5.gif) | ![2022-08-23 15 03 18](https://user-images.githubusercontent.com/14988353/186074119-803cd21e-5343-4a99-9d68-52d99af2af9c.gif) |

Reply:
Interesting it was limited to just this PR for you. I saw it on trunk as well. I'll reset my local dev environment to get a clean slate and dig deeper. Thanks for reporting this 👍

Reply:
I've pushed a refactor that has the block selection working for me again: https://github.com/WordPress/gutenberg/pull/43497/commits/91799d2df72a95d1b7149ec9fe6e4f5efaea5823.

I think you are on the right track regarding the spreading of the `blockProps`. The resulting changeset here is a bit smaller than the first iteration. So I'll call that a win for now.